### PR TITLE
Apply chain id

### PR DESCRIPTION
**Pull request by minsulee2**
Chages:
- Apply chain id
- Did not apply chain id on test codes
- Apply network id as well
- Check network id validity
- Fix some indentations

### CODE REVIEW

File: genesis-configs/testnet/genesis_token.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "bridge": {
     "ETH": {
       "0x3A810ff7211b40c4fA76205a14efe161615d0385": {
-        "network_id": "1",
+        "network_id": 2,
```
Comment (liayoo) [inline genesis-configs/testnet/genesis_token.json:8]: Could you change this to 1 (number)? This indicates the Ethereum mainnet.

File: common/constants.js [hunk 2]
```diff
@@ -110,6 +110,8 @@ const SERVICE_TREE_SIZE_BUDGET = SERVICE_STATE_BUDGET * MAX_STATE_TREE_SIZE_PER_
 const APPS_TREE_SIZE_BUDGET = APPS_STATE_BUDGET * MAX_STATE_TREE_SIZE_PER_BYTE;
 const FREE_TREE_SIZE_BUDGET = FREE_STATE_BUDGET * MAX_STATE_TREE_SIZE_PER_BYTE;
 const STATE_GAS_COEFFICIENT = 1;
+const NETWORK_ID = GenesisParams.blockchain.NETWORK_ID;
+const CHAIN_ID = GenesisParams.blockchain.CHAIN_ID;
```
Comment (liayoo) [inline common/constants.js:114]: I think these are already being exported in L905?

File: genesis-configs/base/genesis_token.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "bridge": {
     "ETH": {
       "0xB16c0C80a81f73204d454426fC413CAe455525A7": {
-        "network_id": "3",
+        "network_id": 0,
```
Comment (liayoo) [inline genesis-configs/base/genesis_token.json:8]: Could you change this to the number 3? (Ethereum ropsten)

File: json_rpc/index.js [hunk 4]
```diff
@@ -374,35 +375,39 @@ module.exports = function getMethods(node, p2pServer, minProtocolVersion, maxPro
...
+      done(null, addProtocolVersion({ result: peerCount }));
     },
 
-    net_syncing: function(args, done) {
+    net_syncing: function (args, done) {
       // TODO(liayoo): Return { starting, latest } with block numbers
       // if the node is currently syncing.
       done(null, addProtocolVersion(
-          {result: p2pServer.node.state === BlockchainNodeStates.SYNCING}));
+        { result: p2pServer.node.state === BlockchainNodeStates.SYNCING }));
```
Comment (platfowner) [inline json_rpc/index.js:392]: minor: 4 space indentation is correct as it's from line wrapping?

alternatives?:
```
done(..., addProtocolVersion({
  result: ...
}));
```

File: p2p/util.js [hunk 5]
```diff
@@ -113,6 +115,16 @@ function checkTimestamp(timestamp) {
   }
 }
 
+function isValidNetworkId(networkId) {
+  if (!CommonUtil.isNumber(networkId)) {
```
Comment (platfowner) [inline p2p/util.js:119]: minor: one line expression? ` return CommonUtil.isNumber(...) && networkId === NETWORK_ID;`
or just `return networkId === NETWORK_ID;`?

File: p2p/index.js [hunk 6]
```diff
@@ -271,9 +273,16 @@ class P2pClient {
     const LOG_HEADER = 'setClientSidePeerEventHandlers';
     socket.on('message', (message) => {
       const parsedMessage = JSON.parse(message);
+      const networkId = _.get(parsedMessage, 'networkId');
```
Comment (platfowner) [inline p2p/index.js:276]: Question: Where is the network id payloaded?

File: genesis-configs/testnet/genesis_token.json (hunk 1 above)
Comment (minsulee2) [reply genesis-configs/testnet/genesis_token.json]: Sorry for confusion! Fixed!

File: common/constants.js (hunk 2 above)
Comment (minsulee2) [reply common/constants.js]: Yes yes I missed it. Fixed!

File: genesis-configs/base/genesis_token.json (hunk 3 above)
Comment (minsulee2) [reply genesis-configs/base/genesis_token.json]: Done!

File: p2p/util.js (hunk 5 above)
Comment (minsulee2) [reply p2p/util.js]: Seems `return networkId === NETWORK_ID;` enough. updated!

File: json_rpc/index.js (hunk 4 above)
Comment (minsulee2) [reply json_rpc/index.js]: Yes yes!! Fixed!

File: p2p/index.js (hunk 6 above)
Comment (minsulee2) [reply p2p/index.js:276]: It is payloaded in encapsulateMessage function at p2p/util.js line 99.

**Comment by minsulee2** [comment]
PTAL. All addressed!

**Review by platfowner — COMMENTED** [review]
LGTM
Please wait for @liayoo 's approval. Thanks!

**Comment by liayoo** [comment]
Could you update the tests as well? :)

**Comment by minsulee2** [comment]
@liayoo Broken tests have just been fixed!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!!

File: p2p/index.js (hunk 6 above)
Comment (platfowner) [reply p2p/index.js:276]: Got it. Thanks!

**Review by platfowner — APPROVED** [review]
LGTM

**Comment by minsulee2** [comment]
Thanks! merging now!

**Pull request merged**